### ISSUE TITLE
TextChannelConverter says channel cannot be found when it can but it's not a TextChannel

### ISSUE THREAD

**Issue by Borketh** [nextcord/nextcord#1216]
### Summary

see title

### Reproduction Steps

Call any command with an argument of type `TextChannelConverter` on the ID of a `ForumChannel` or similar.

### Expected Results

Some sort of error that is more relevant and informative, such as a `TypeError` or `InvalidArgument` that tells me that the channel *can* be found but is not a `TextChannel` and is unsuitable for use in this command.

### Actual Results

`BadArgument` is thrown with the description `"Channel {id} not found."`

### Intents

all

### System Information

- Python v3.12.7-final
- nextcord v2.6.0-final
- aiohttp v3.10.9
- system info: `Linux 6.11.2-1-cachyos-eevdf-lto #1 SMP PREEMPT_DYNAMIC Sat, 05 Oct 2024 06:39:17 +0000`

### Checklist

- [X] I have searched the open issues for duplicates.
- [X] I have shown the entire traceback, if possible.
- [X] I have removed my token from display, if visible.

**Comment by EmmmaTech** [comment]
I agree with this, but this seems more like a feature change and less like a bug report. this was an intentional change in the dpy era, specifically v1.5.

**Comment by Borketh** [comment]
I agree, although I don't seem to be able to change labels myself. Might be gh mobile moment.

**Comment by EmmmaTech** [comment]
my bad, I forgot to change labels! you can't change labels unless you're a collaborator on the repo, but when you make an issue, templates allow you to automatically add a label.
